test/performance/operf.go

```diff
@@ -0,0 +1,634 @@
+				return time.Now().Unix(), time.Now().Unix()
+			}
+		}
+	}
+	tsEndTest := time.Now().Unix()
+
+	fmt.Println("Times report")
+	fmt.Println(strings.Repeat("-", 80))
+	fmt.Printf("Start test      : %s\n", time.Unix(tsStartTest, 0))
+	fmt.Printf("End test        : %s\n", time.Unix(tsEndTest, 0))
+
+	return tsStartTest, tsEndTest
+}
+func usageTest(c *kubernetes.Clientset) (int64, int64) {
+	tsStartTest := time.Now().Unix()
+	// Create PVCs
+	fmt.Printf("%d PVCs created\n", createPVC(c, *namePattern, *numberInstances))
+	tsPVCSCreated := time.Now().Unix()
+
+	// Use PVCs
+	fmt.Printf("%d Pods created\n", createPods(c, *namePattern, *numberInstances))
+	tsPVCSUsed := time.Now().Unix()
+
+	//Wait for Pods Running
+	fmt.Printf("%d Pods running\n", waitForPodsRunning(c))
+
+	// Free PVCs
+	if deletePods(c) != nil {
+		fmt.Printf("Error trying to delete test Pods\n")
+	} else {
+		fmt.Printf("test Pods deleted\n")
+	}
+	tsPVCSFree := time.Now().Unix()
+
+	// Delete PVCs
+	if deletePVCs(c) != nil {
+		fmt.Printf("Error trying to delete test PVCs\n")
+	} else {
+		fmt.Printf("test PVCs deleted\n")
+	}
+	tries := 0
+	waitInterval := 10 * time.Second
+	for notDeleted := true; notDeleted; notDeleted = !pvcsDeleted(c) {
+		fmt.Printf("Waiting for PVCS clean \n")
+		time.Sleep(waitInterval)
+		tries++
+		if tries == 20 {
+			fmt.Printf("Timeout (%d seconds)waiting for PVCS cleaning", waitInterval)
+			break
+		}
+	}
+	tsEndTest := time.Now().Unix()
+
+	fmt.Println("Times report")
+	fmt.Println(strings.Repeat("-", 80))
+	fmt.Printf("Start test      : %s\n", time.Unix(tsStartTest, 0))
+	fmt.Printf("PVCs created    : %s\n", time.Unix(tsPVCSCreated, 0))
+	fmt.Printf("PVCs utilization: %s\n", time.Unix(tsPVCSUsed, 0))
+	fmt.Printf("PVCs Free       : %s\n", time.Unix(tsPVCSFree, 0))
+	fmt.Printf("PVCs deleted    : %s\n", time.Unix(tsEndTest, 0))
+	fmt.Printf("End test        : %s\n", time.Unix(tsEndTest, 0))
+
+	fmt.Printf("Usage report: %d pods in namespace %s using %d PVCs with Storage Class <%s>\n",
+		*numberInstances,
+		*testNamespace,
+		*numberInstances,
+		*testStorageClassName)
+
+	return tsStartTest, tsEndTest
+}
+
+func createPVC(c *kubernetes.Clientset, PVCName string, PVCsNumber int) int {
+	var pvc = corev1.PersistentVolumeClaim{}
+	var numPVCsCreated = 0
+
+	for i := 0; i < PVCsNumber; i++ {
+		pvc = corev1.PersistentVolumeClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("%s-%d", PVCName, i),
+				Namespace: *testNamespace,
+				Labels:    PVCLabels,
+			},
+			Spec: corev1.PersistentVolumeClaimSpec{
+				AccessModes:      []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce},
+				StorageClassName: testStorageClassName,
+				Resources: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{
+						corev1.ResourceName(corev1.ResourceStorage): resource.MustParse("128Mi"),
+					},
+				},
+			},
+		}
+		_, err := c.CoreV1().PersistentVolumeClaims(*testNamespace).Create(context.TODO(), &pvc, metav1.CreateOptions{})
```

Comment:
Proper context should be created and passed on to all the methods rather than `context.TODO()`.